soh/src/overlays/actors/ovl_En_Dns/z_en_dns.c

```diff
@@ -185,6 +204,13 @@ void EnDns_ChangeAnim(EnDns* this, u8 index) {
 
 /* Item give checking functions */
 
+u32 EnDns_RandomizerPurchaseableCheck(EnDns* this) {
+    if (gSaveContext.rupees < this->dnsItemEntry->itemPrice || gSaveContext.scrubsPurchased[this->scrubIdentity.scrubId] == 1) {
+        return 0;
+    }
+    return 4;
```

Comment:
assuming this is `4` to be `0b100` for some bitwise stuff in the purchaseableCheck

Reply:
It's used here, with 0,1, and 3 being failure states. Probably could be an enum but I'm not sure what the detailed difference between them all is
<img width="342" alt="Screen Shot 2022-08-25 at 1 36 49 PM" src="https://user-images.githubusercontent.com/7316699/186742780-33ee244b-a442-428f-bd94-91f1445fd490.png">

Reply:
![image](https://user-images.githubusercontent.com/70942617/186760257-eba61fa7-927d-48cf-9a88-83b080427160.png)
![image](https://user-images.githubusercontent.com/70942617/186760332-1258b269-8947-4bad-bd19-9c15fe16e0e2.png)
![image](https://user-images.githubusercontent.com/70942617/186760382-1c591fee-bc39-458a-8ef5-e3bafe7b4fce.png)

oddest thing in that code is it seems both `2` and `4` get us to the successful purchase.

Reply:
anyways, feels like a magic number but it's cool for now